### PR TITLE
Tftp detection

### PR DESCRIPTION
**Pull request by dkgroot**
Reverse the logic when detecting tftpserver existence and path
 - write a sentinel probe file to the known/possible distro tftp directories
 - try fetching it through tftp
 - if we manage to fetch the file, we have the correct path
 - remove sentinel probe file

sccpManClasses/extconfigs.class.php : When creating directories in tftpboot directory we should not set write permissions, using 0755 instead.

### CODE REVIEW

File: sccpManTraits/helperFunctions.php
Comment (steve-lad) [inline sccpManTraits/helperFunctions.php:190]: Need to send an ack here or the server will try resending until timeout

File: install.php
Comment (steve-lad) [inline install.php]: file_put_contents($tempFile, $remoteFileContent);

is simpler?

**Review by steve-lad — CHANGES_REQUESTED** [review]
Looks good with comments

File: sccpManTraits/helperFunctions.php
Comment (dkgroot) [reply sccpManTraits/helperFunctions.php:190]: Aah ok, i guess i overlooked that, you tried telling me yesterday :-)

File: install.php
Comment (dkgroot) [reply install.php]: I am a C guy, and it shows :-) Will use your recommendation.

**Comment by dkgroot** [comment]
Thanks for the review !

**Review by steve-lad — APPROVED** [review]
Thanks

**Comment by dkgroot** [comment]
LIttle heads up regarding the 0755 change in the extconfigs.class.php.

My tftp server runs as "tftpd:tftpd" but the directory is owned by "asterisk:"asterisk" so that sccp_manager / freepbx can write to that directory. I think this should be assumed as normal operation with least permission & surprise. Because my tftp server does not own this directory, it's using this directory with 'other' permissions. When creating directories we should give "group" and "other" only "r+X" (5) and when writing files we should give those files only "r" (4). SCCP phone have no business writing files, so having write access through tftp should not be a requirement.

Note: SPA phones can write their config template to a tftp server when requested to do so, this might be an exception. But in most cases this is a one time action.

**Pull request merged**
